workshops/check.py

```diff
@@ -24,10 +10,14 @@
 import os
 import re
 import logging
-
 import yaml
 from collections import Counter
 
+try:  # Hack to make codebase compatible with python 2 and 3
+    basestring
+except NameError:
+    basestring = str
```

Comment:
What I'm used to doing is:

    try:
        our_basestring = basestring
    except NameError:
        our_basestring = str

and then using `our_basestring` everywhere, rather than giving readers the possibly-mistaken impression that `basestring` in our code is actually what they think of as `basestring`.  It's a minor (really minor) nit.